### PR TITLE
Data loss protection

### PR DESCRIPTION
**Pull request by btcontract**
Updated to match changes in `master`

### CODE REVIEW

File: eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala [hunk 1]
```diff
@@ -1142,16 +1142,22 @@ class Channel(val nodeParams: NodeParams, wallet: EclairWallet, remoteNodeId: Pu
...
 
-    case Event(INPUT_DISCONNECTED, _) => stay // we are disconnected, but it doesn't matter anymoer
+    case Event(INPUT_DISCONNECTED, _) => stay // we are disconnected, but it doesn't matter anymore
   })
 
   when(OFFLINE)(handleExceptions {
     case Event(INPUT_RECONNECTED(r), d: HasCommitments) =>
       forwarder ! r
+
+      val yourLastPerCommitmentSecret = d.commitments.remotePerCommitmentSecrets.lastIndex.flatMap(d.commitments.remotePerCommitmentSecrets.getHash).getOrElse(Sphinx.zeroes(32))
```
Comment (pm47) [inline eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala]: Where did you get the default value `Sphinx.zeroes(32)`? Can't find it in the spec

Comment (pm47) [reply eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala]: nitpick: `yourLastPerCommitmentSecret ` should be of type `Scalar`

**Comment by pm47** [comment]
Also a rebase would be nice :-)

File: eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala (hunk 1 above)
Comment (btcontract) [reply eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala]: I'm sure it was there in spec but can't find it either. What should be there if `lastIndex` is `None`?

Comment (pm47) [reply eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala]: if `lastIndex`=`None` then that's our first commitment, and we shouldn't send anything maybe?

**Comment by btcontract** [comment]
Please do take over if this is urgent. It's in my plans to continue but currently I'm overwhelmed by work on lnwallet.

**Comment by btcontract** [comment]
I'm ready to resume on this unless you've taken over already, please let me know.

**Pull request closed without merging**